### PR TITLE
[stdlib] rewrite optional handling using map

### PR DESCRIPTION
**Pull request by Nirma**
I thought that rewriting the optional handling using map would make it a little bit
cleaner to read.

Hope its ok!

### CODE REVIEW

**Comment by Nirma** [comment]
@gribozavr Is this kind of a change too much of a nitpick? 
Sorry if it is. 🙇

File: stdlib/public/core/String.swift [hunk 1]
```diff
@@ -321,11 +321,10 @@ extension String {
     let (stringBufferOptional, _) =
         _StringBuffer.fromCodeUnits(input, encoding: encoding,
             repairIllFormedSequences: false)
-    if let stringBuffer = stringBufferOptional {
-      return String(_storage: stringBuffer)
-    } else {
+    guard let stringBuffer = stringBufferOptional else {
       return nil
     }
+    return String(_storage: stringBuffer)
```
Comment (gribozavr) [inline stdlib/public/core/String.swift]: I think this would actually be better as `return stringBufferOptional.map { String(_storage: $0) }`.

**Comment by Nirma** [comment]
@gribozavr Thanks! That is a lot cleaner, I updated the pull request.

How does it look now?

**Comment by gribozavr** [comment]
LGTM, thanks!

@swift-ci Please smoke test

**Pull request merged**